helix-term/src/commands.rs

```diff
@@ -2915,6 +2920,66 @@ fn goto_prev_diag(cx: &mut Context) {
     goto_pos(editor, pos);
 }
 
+fn goto_first_change(cx: &mut Context) {
+    goto_first_hunk_impl(cx, false);
+}
+
+fn goto_last_change(cx: &mut Context) {
+    goto_first_hunk_impl(cx, true);
+}
+
+fn goto_first_hunk_impl(cx: &mut Context, reverse: bool) {
+    let editor = &mut cx.editor;
+    let (_, doc) = current!(editor);
+    if let Some(handle) = doc.diff_handle() {
+        let hunk = {
+            let hunks = handle.hunks();
+            let idx = if reverse {
+                hunks.len().saturating_sub(1)
+            } else {
+                0
+            };
+            hunks.nth_hunk(idx)
+        };
+        if hunk != Hunk::NONE {
+            let pos = doc.text().line_to_char(hunk.after.start as usize);
+            goto_pos(editor, pos)
+        }
+    }
+}
+
+fn goto_next_change(cx: &mut Context) {
+    goto_next_change_impl::<false>(cx)
+}
+
+fn goto_prev_change(cx: &mut Context) {
+    goto_next_change_impl::<true>(cx)
+}
+
+fn goto_next_change_impl<const REVERSE: bool>(cx: &mut Context) {
+    let editor = &mut cx.editor;
+    let (view, doc) = current!(editor);
+
+    let cursor_pos = doc
+        .selection(view.id)
+        .primary()
+        .cursor(doc.text().slice(..));
+
+    let cursor_line = doc.text().char_to_line(cursor_pos);
+
+    let next_changed_line = doc.diff_handle().and_then(|diff_handle| {
+        let hunks = diff_handle.hunks();
+        let hunk_idx = hunks.next_hunk::<REVERSE>(cursor_line as u32)?;
+        let line = hunks.nth_hunk(hunk_idx).after.start;
+        Some(line)
+    });
+
+    if let Some(next_change_line) = next_changed_line {
+        let pos = doc.text().line_to_char(next_change_line as usize);
+        goto_pos(cx.editor, pos)
+    }
```

Comment:
[`goto_ts_object_impl`](https://github.com/helix-editor/helix/blob/607c74efde40812caa6379a05fcb28f259ea2c8e/helix-term/src/commands.rs#L4404-L4443) is the 'state of the art' on these kinds of jumps:

* `cx.count()` should be taken into account so that you can jump over multiple hunks.
* setting `cx.editor.last_motion` to the motion allows you to repeat the motion with `A-.`.
* `Selection::transform` lets you operate on all cursors at once. I'm not sure how useful that would be for this motion but I don't think we should limit ourselves to just the primary cursor.
* Negotiating the head, anchor and `editor.mode` makes these extend in select mode.

Reply:
Apologies I should have read into existing motions more. I updated the motion to behave correctly.